src/dialogs/generic/show-dialog-box.ts

```diff
@@ -0,0 +1,61 @@
+import { fireEvent } from "../../common/dom/fire_event";
+
+export interface AlertDialogParams {
+  title?: string;
+  text?: string;
+  dismissText?: string;
+  confirm?: (out?: string) => void;
+}
+
+export interface ConfirmationDialogParams extends AlertDialogParams {
+  confirmText?: string;
+  cancel?: () => void;
+}
+
+export interface PromptDialogParams extends ConfirmationDialogParams {
+  inputType?: string;
+  inputLabel?: string;
+}
+
+export interface GenericDialogParams
+  extends ConfirmationDialogParams,
+    PromptDialogParams {
+  confirmation: boolean;
+  prompt: boolean;
+}
+
+export const loadGenericDialog = () =>
+  import(/* webpackChunkName: "confirmation" */ "./dialog-box");
+
+export const showAlertDialog = (
+  element: HTMLElement,
+  dialogParams: AlertDialogParams
+): void => {
+  fireEvent(element, "show-dialog", {
```

Comment:
create one functions that fires the event, that takes an argument for the config.
```typescript
showDialog(dialogParams) {
  fireEvent(element, "show-dialog", {
    dialogTag: "dialog-box",
    dialogImport: loadGenericDialog,
    dialogParams,
  });
}
```